pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/plan/server/ServerPlanRequestVisitor.java

```diff
@@ -156,15 +156,15 @@ public Void visitJoin(JoinNode node, ServerPlanRequestContext context) {
       if (visit(left, context)) {
         PipelineBreakerResult pipelineBreakerResult = context.getPipelineBreakerResult();
         int resultMapId = pipelineBreakerResult.getNodeIdMap().get(right);
-        List<TransferableBlock> transferableBlocks =
-            pipelineBreakerResult.getResultMap().getOrDefault(resultMapId, Collections.emptyList());
+        List<MseBlock> blocks = pipelineBreakerResult.getResultMap().getOrDefault(resultMapId, Collections.emptyList());
         List<Object[]> resultDataContainer = new ArrayList<>();
         DataSchema dataSchema = right.getDataSchema();
```

Comment:
Could this loop receive column-based block ?

Reply:
I think this comment should be a bit below, right? I think you meant:

```java
        for (TransferableBlock block : transferableBlocks) {
          if (block.getType() == DataBlock.Type.ROW) {
            resultDataContainer.addAll(block.getContainer());
          }
        }
```

Which is now:

```java
        for (MseBlock block : blocks) {
          if (block.isData()) {
            resultDataContainer.addAll(((MseBlock.Data) block).asRowHeap().getRows());
          }
        }
```

I don't think we could receive column base blocks there. In fact I didn't see a single place where column blocks were used. Specifically in this case, I hope we weren't receiving any columnar block, otherwise we would just skip it and return an incorrect value